src/sources/udp.rs

```diff
@@ -0,0 +1,220 @@
+use crate::{
+    event::{self, Event},
+    topology::config::{DataType, GlobalOptions, SourceConfig},
+};
+use codec::{self, BytesDelimitedCodec};
+use futures::{future, stream::iter_ok, sync::mpsc, Future, Sink, Stream};
+use serde::{Deserialize, Serialize};
+use std::{io, net::SocketAddr};
+use string_cache::DefaultAtom as Atom;
+use tokio::{
+    self,
```

Comment:
this `self` can be dropped since we can just access tokio directly in rust 2018 edition.

Reply:
Indeed